src/sync/partial_sync.cpp

```diff
@@ -241,32 +335,95 @@ RowExpr write_subscription(std::string const& object_type, std::string const& na
+                                                              "a different type is not allowed: %1 vs. %2",
+                                                              existing_matching_property, matches_property));
+            }
+
+            table->set_string(columns.query, row_ndx, query);
+            if (time_to_live_ms) {
+                table->set_int(columns.time_to_live, row_ndx, time_to_live_ms.value());
+            }
+            else {
+                table->set_null(columns.time_to_live, row_ndx);
+            }
+        }
+        else {
+            StringData existing_query = table->get_string(columns.query, row_ndx);
+            if (existing_query != query)
+                throw ExistingSubscriptionException(util::format("An existing subscription exists with the same name, "
+                                                                 "but a different query ('%1' vs '%2').",
+                                                                 existing_query, query));
+
+            StringData existing_matches_property = table->get_string(columns.matches_property_name, row_ndx);
+            if (existing_matches_property != matches_property)
+                throw ExistingSubscriptionException(util::format("An existing subscription exists with the same name, "
+                                                                 "but a different result type ('%1' vs '%2').",
+                                                                 existing_matches_property, matches_property));
+        }
+
+    }
+    else {
+        // No existing subscription was found. Create a new one
         row_ndx = sync::create_object(group, *table);
         table->set_string(columns.name, row_ndx, name);
         table->set_string(columns.query, row_ndx, query);
         table->set_string(columns.matches_property_name, row_ndx, matches_property);
+        table->set_timestamp(columns.created_at, row_ndx, now);
+        if (time_to_live_ms) {
+            table->set_int(columns.time_to_live, row_ndx, time_to_live_ms.value());
+        }
+        else {
+            table->set_null(columns.time_to_live, row_ndx);
+        }
+    }
+
+    // Always set updated_at/expires_at when a subscription is touched, no matter if it is new, updated or someone just
+    // resubscribes.
+    table->set_timestamp(columns.updated_at, row_ndx, now);
+    if (table->is_null(columns.time_to_live, row_ndx)) {
+        table->set_null(columns.expires_at, row_ndx);
+    }
+    else {
+        table->set_timestamp(columns.expires_at, row_ndx, calculate_expiry_date(now, table->get_int(columns.time_to_live, row_ndx)));
     }
-    return table->get(row_ndx);
+
+    // Fetch subscription first and return it. Cleanup needs to be performed after as it might delete subscription
+    // causing the row_ndx to change.
+    RowExpr subscription = table->get(row_ndx);
```

Comment:
This needs to be `Row`. Storing a `RowExpr` is identical to just storing the row index itself.